webhooks/virtualmachine/v1alpha1/mutation/virtualmachine_mutator.go

```diff
@@ -154,6 +186,48 @@ func SetDefaultPowerState(
 	return false
 }
 
+// ResolveImageName mutates the vm.spec.imageName if it's not set to a vmi name
+// and there is a single namespace or cluster scope image with that status name.
+func ResolveImageName(
+	ctx *context.WebhookRequestContext,
+	c client.Client,
+	vm *vmopv1.VirtualMachine) (bool, error) {
+	// Return early if the VM image name is empty or already set to a vmi name.
+	imageName := vm.Spec.ImageName
+	if imageName == "" || !lib.IsWCPVMImageRegistryEnabled() ||
+		strings.HasPrefix(imageName, "vmi-") {
+		return false, nil
+	}
+
+	// Check if a single namespace scope image exists by the status name.
+	vmiList := &vmopv1.VirtualMachineImageList{}
+	if err := c.List(ctx, vmiList, client.InNamespace(vm.Namespace),
+		client.MatchingFields{
+			"status.imageName": imageName,
+		},
+	); err != nil {
+		return false, err
+	}
+	if len(vmiList.Items) == 1 {
```

Comment:
in the case where 2 vmi and 1 cvmi has the same friendly name, we return cvmi. Is this the behavior we want?

Reply:
It's a rare case where the same image needs to be pushed into two CLs associated with the namespace, as well as another CL associated with the cluster. However, if that happens, I think it's acceptable to return the cvmi as it's determined in the cluster scope.